### PR TITLE
Log streaming resource cleanup 

### PR DESCRIPTION
**Pull request by Aayyush**

### CODE REVIEW

File: server/events/project_finder.go [hunk 1]
```diff
@@ -168,6 +181,53 @@ func (p *DefaultProjectFinder) filterToFileList(log logging.SimpleLogging, files
...
+	if err != nil {
+		return pullInfoList, errors.Wrap(err, "retrieving working dir")
+	}
+
+	modifiedFiles, err := p.VCSClient.GetModifiedFiles(repo, pull)
+	if err != nil {
+		return pullInfoList, errors.Wrap(err, "retrieving modified files")
+	}
+
+	hasRepoCfg, err := p.ParserVarlidator.HasRepoCfg(repoDir)
```
Comment (nishkrishnan) [inline server/events/project_finder.go:196]: Hmm I'm not sure this is the best approach to get the list of projects for a given PR.  I think you can just leverage the db here:
https://github.com/lyft/atlantis/blob/release-v0.17.1-lyft.1/server/events/db/boltdb.go#L380

It should give you all the projects in a given pull request. Then you shouldn't need this whole function at all.

Comment (msarvar) [reply server/events/project_finder.go:196]: +1 on using `Db.GetPullStatus(pull)`.

File: server/events/pull_closed_executor.go [hunk 2]
```diff
@@ -73,6 +75,24 @@ var pullClosedTemplate = template.Must(template.New("").Parse(
...
 func (p *PullClosedExecutor) CleanUpPull(repo models.Repo, pull models.PullRequest) error {
+	pullStatus, err := p.DB.GetPullStatus(pull)
+	if err != nil {
+		// Log and continue to clean up other resources.
+		p.Logger.Err("retrieving pull status: %s", err)
+	}
+
+	if pullStatus != nil {
+		for _, project := range pullStatus.Projects {
+			// TODO: Set projectName to "<dir>/<workspace>" when project name is not set.
```
Comment (nishkrishnan) [inline server/events/pull_closed_executor.go:86]: Mind creating a jira for this and then putting the ticket number here? will make it easier to track the places we need to support this.

**Review by nishkrishnan — APPROVED** [review]
Nice work!

File: server/events/pull_closed_executor.go (hunk 2 above)
Comment (Aayyush) [reply server/events/pull_closed_executor.go]: Done!

**Review by msarvar — APPROVED** [review]
good job

**Pull request merged**